enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+capability is not enabled
+OR
+4b) during upgrade, the existing registry resources continue to be applied(because we have a rule
+that says once a resource is applied, we will continue to reconcile it in future versions), but
+new registry resources added to the payload are not applied because they are filtered due to the
+registry capability not being enabled, resulting in critical registry resources not being applied
+and the registry potentially breaking.
+
+This scenario is avoided by specifying that once a resource is applied to a cluster, it will
+always be reconciled in future versions, and that if any part of a capability is present,
+the entire capability will be present.
+
+This means the CVO must apply the following logic:
+
+1) for the new payload, find all resources that are filtered by the current cluster config
+2) check if any of those resources are already applied to the cluster by explicilty
+checking what is on the cluster already
+3) for any resource that is being filtered, but already exists in the cluster apply that
+resource and implicitly enable any capabilities it is a part of (can be reported in Status)
+which may mean additional resources get applied.
+
+
+Behavior Axioms:
+1) Cluster admin can explicitly opt into, or out of, any capability at install time.
+2) Cluster admin can indicate that all capabilities not explicitly opted into/out of(via (1)), should
+   either be enabled, or not enabled.  This choice will also apply to new capabilities introduced in upgrades.
+3a) Cluster admin cannot disable an enabled capability post-install.  They can be enabled post-install.  (it is
+too difficult to safely/fully clean up the cluster when disabling a capability)
+3b) Once the CVO has applied/reconciled a resource, it must continue to do so even if future upgrades
+mean that resource would nominally be excluded (such as because the capability it is associated
+with has changed to a capability that is disabled on the cluster).
+4) Cluster Profiles take precedence over a resource's capability association and inclusion.  If the
+   cluster's profile doesn't include a particular resource, enabling the capability won't add that
+   resource to the cluster.  If the cluster profile does include it, it can still be excluded via
+   the capabilities mechanism.
+5) Individual capabilities/resources cannot specify whether they should be included or excluded
+   by default.  If a resource is in the payload, it is included by default, unless the cluster
+   admin has made the choice to exclude all "optional" content by default.
+6) If any resource from a capability is applied to the cluster, the other resources in the
+capability must also be applied to the cluster, regardless of the expressed configuration.  This
+may result in a capability the admin explicitly disabled, being enabled as part of an upgrade.  E.g.
+if a resource that was already applied to the cluster becomes part of a capability that the admin
+had explicitly disabled.
```

Comment:
https://github.com/openshift/enhancements/blob/master/enhancements/update/object-removal-manifest-annotation.md

basically you can annotate a resource to tell the CVO "just delete this thing if it exists on the cluster"

but if you don't do that and just delete the resource, yes it would be orphaned on the cluster.  Or you could have your operator logic itself clean it up.